### PR TITLE
Fixed colorization for GridGenerator::half_hyper_shell

### PR DESCRIPTION
**Pull request by masterleinad**
The documentation for GridGenerator::half_hyper_shell was kind of misleading. Actually, there was no colorization implemented. This should be fixed now.

### CODE REVIEW

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -506,8 +506,8 @@ namespace GridGenerator
    * computed adaptively such that the resulting elements have the least
    * aspect ratio.
    *
-   * If colorize is set to true, the inner, outer, left, and right boundary
-   * get indicator 0, 1, 2, and 3, respectively. Otherwise all indicators are
+   * If colorize is set to true, the inner, outer and equatorial boundary
+   * get indicator 0, 1, and 2, respectively. Otherwise all indicators are
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: I'm not sure what you mean by "equatorial". Do you mean the "radial" boundaries? (That may not be clearer either -- maybe more words are necessary.)

File: source/grid/grid_generator.cc [hunk 2]
```diff
@@ -3091,7 +3089,32 @@ namespace GridGenerator
...
+            if (cell->at_boundary(i))
+              {
+                const typename Triangulation<3>::face_iterator face
+                  = cell->face(i);
 
+                const Point<3> face_center (face->center());
+                if (std::abs((face_center(0)-center(0)) > 1.e-3))
+                  {
+                    if (std::abs((face_center-center).norm()-inner_radius) <
+                        std::abs((face_center-center).norm()-outer_radius))
```
Comment (bangerth) [inline source/grid/grid_generator.cc]: I find these nested ifs hard to follow. Can you add a comment how they are supposed to work?

File: include/deal.II/grid/grid_generator.h (hunk 1 above)
Comment (masterleinad) [inline include/deal.II/grid/grid_generator.h]: Essentially, I mean by equatorial the boundary at x=0. I can rephrase that.

File: include/deal.II/grid/grid_generator.h [hunk 3]
```diff
@@ -506,9 +506,9 @@ namespace GridGenerator
    * computed adaptively such that the resulting elements have the least
    * aspect ratio.
    *
-   * If colorize is set to true, the inner, outer, left, and right boundary
-   * get indicator 0, 1, 2, and 3, respectively. Otherwise all indicators are
-   * set to 0.
+   * If colorize is set to true, the inner, outer and equatorial (i.e. where 
+   * the first component is zero) boundary get indicator 0, 1, and 2, respectively.
```
Comment (bangerth) [inline include/deal.II/grid/grid_generator.h]: I think the term we were apparently both looking for that would have fit is "equatorial plane". Except, of course, that is almost universally considered the plane where `z==0` (or in 2d, `y==0`). So maybe *any* reference to the word "equator" is confusing here. Maybe just say "inner, outer, and the part of the boundary where $x=0$, get indicator 0, 1, and 2, respectively."?

Comment (masterleinad) [inline include/deal.II/grid/grid_generator.h]: I am fine with that as well.

**Comment by bangerth** [comment]
Great, thanks! (I guess a testcase would not have hurt, in hindsight. If you feel like it, we'd accepted a patch ;-)
